File: Examples/UI/WinForm/VB.NET/MainForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Reflection
Imports System.Diagnostics
Imports System.Drawing
Imports System.Collections
Imports System.Collections.Specialized
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.IO

Imports Nevron.Globalization
Imports Nevron.UI
Imports Nevron.UI.WinForm
Imports Nevron.UI.WinForm.Controls
Imports Nevron.UI.WinForm.Docking
Imports Nevron.Examples.Framework.WinForm
Imports Nevron.GraphicsCore

Namespace Nevron.Examples.UI.WinForm
	''' <summary>
	''' Summary description for MainForm.
	''' </summary>
	Public Class MainForm
		Inherits NExamplesForm
		#Region "Constructor"

		Public Sub New()
			InitializeComponent()

			InitFromConfig(New NUIExamplesConfig())

			CaptionImage = Config.m_FormCaptionImage
			Size = New Size(800, 600)
		End Sub

		Shared Sub New()
			Dim t As Type = GetType(MainForm)
			Dim path As String = "Nevron.Examples.UI.WinForm.Resources"

			StandardImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Silver, "Standard.bmp", path)
			ActionImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "Action.bmp", path)
			LayoutImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "Layout.bmp", path)
			ToolsImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "Tools.bmp", path)
			ViewImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "View.bmp", path)
			TestImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "Test.bmp", path)
			FormatImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "Format.bmp", path)
			MiscImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "Misc.bmp", path)
			DockingImages = NResourceHelper.ImageListFromBitmap(GetType(MainForm), New Size(16, 16), Color.Magenta, "Docking.bmp", path)
		End Sub


		#End Region

		#Region "Overrides"

		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not components Is Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		Public Overrides Sub LoadNavigationTree()
			Dim tree As TreeView = NavigationTreeControl.NavigationTree
			tree.Nodes.Clear()

			Dim rootFolder As NExampleFolderEntity = NExamplesHelper.LoadExamplesTree(Config.EmbeddedResourcesAssembly, Config.TreeResource, Config.TreeResourcePath)
			tree.Nodes.Add(LoadTreeFolder(String.Empty, rootFolder))
		End Sub


		#End Region

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			' 
			' MainForm
			' 
			Me.AutoScaleBaseSize = New System.Drawing.Size(5, 14)
			Me.ClientSize = New System.Drawing.Size(792, 566)
			Me.Font = New System.Drawing.Font("Tahoma", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, (CByte(204)))
			Me.Name = "MainForm"
			Me.Text = "MainForm"

		End Sub
		#End Region

		#Region "Fields"

		Private components As System.ComponentModel.Container = Nothing

		Friend Shared StandardImages As ImageList
		Friend Shared ActionImages As ImageList
		Friend Shared LayoutImages As ImageList
		Friend Shared ToolsImages As ImageList
		Friend Shared ViewImages As ImageList
		Friend Shared FormatImages As ImageList
		Friend Shared MiscImages As ImageList
		Friend Shared DockingImages As ImageList
		Friend Shared TestImages As ImageList

		#End Region

		#Region "Main"

		''' <summary>
		''' The main entry point for the application.
		''' </summary>
		<STAThread> _
		Shared Sub Main()
			Application.EnableVisualStyles()
			Application.DoEvents()
			Dim form As MainForm = New MainForm()

			'try
			'{
				Application.Run(form)
			'}
			'catch
'{
'}
		End Sub

		#End Region
	End Class
End Namespace

File: Examples/UI/WinForm/CSharp/MainForm.cs
using System;
using System.Reflection;
using System.Diagnostics;
using System.Drawing;
using System.Collections;
using System.Collections.Specialized;
using System.ComponentModel;
using System.Windows.Forms;
using System.IO;

using Nevron.Globalization;
using Nevron.UI;
using Nevron.UI.WinForm;
using Nevron.UI.WinForm.Controls;
using Nevron.UI.WinForm.Docking;
using Nevron.Examples.Framework.WinForm;
using Nevron.GraphicsCore;

namespace Nevron.Examples.UI.WinForm
{
	/// <summary>
	/// Summary description for MainForm.
	/// </summary>
	public class MainForm : NExamplesForm
	{
		#region Constructor

		public MainForm()
		{
			InitializeComponent();

			InitFromConfig(new NUIExamplesConfig());

			Size = new Size(800, 600);
		}

		static MainForm()
		{
			Type t = typeof(MainForm);
			string path = "Nevron.Examples.UI.WinForm.Resources";

			StandardImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Silver, "Standard.bmp", path);
			ActionImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "Action.bmp", path);
			LayoutImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "Layout.bmp", path);
			ToolsImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "Tools.bmp", path);
			ViewImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "View.bmp", path);
			TestImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "Test.bmp", path);
			FormatImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "Format.bmp", path);
			MiscImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "Misc.bmp", path);
			DockingImages = NResourceHelper.ImageListFromBitmap(typeof(MainForm), new Size(16, 16), Color.Magenta, "Docking.bmp", path);
		}

		#endregion

		#region Overrides

		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		public override void LoadNavigationTree()
		{
			TreeView tree = NavigationTreeControl.NavigationTree;
			tree.Nodes.Clear();

			NExampleFolderEntity rootFolder = NExamplesHelper.LoadExamplesTree(Config.EmbeddedResourcesAssembly, Config.TreeResource, Config.TreeResourcePath);
			tree.Nodes.Add(LoadTreeFolder( string.Empty, rootFolder ));
		}


		#endregion

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			// 
			// MainForm
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 14);
			this.ClientSize = new System.Drawing.Size(792, 566);
			this.Font = new System.Drawing.Font("Tahoma", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(204)));
			this.Name = "MainForm";
			this.Text = "MainForm";

		}
		#endregion

		#region Fields

		private System.ComponentModel.Container components = null;

		internal static ImageList StandardImages;
		internal static ImageList ActionImages;
		internal static ImageList LayoutImages;
		internal static ImageList ToolsImages;
		internal static ImageList ViewImages;
		internal static ImageList FormatImages;
		internal static ImageList MiscImages;
		internal static ImageList DockingImages;
		internal static ImageList TestImages;

		#endregion

		#region Main

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
            Application.EnableVisualStyles();
            Application.DoEvents();
			MainForm form = new MainForm();

			//try
			//{
				Application.Run(form);
			//}
			/*catch
			{
			}*/
		 }

		#endregion
	}
}
